packages/ui/src/components/Dialog/Dialog.tsx

```diff
@@ -125,51 +119,58 @@ DialogClose.displayName = DialogPrimitive.Close.displayName;
 const DialogHeader = ({
   className,
   size,
+  children,
+  padding = true,
   ...props
-}: React.HTMLAttributes<HTMLDivElement> & SizeProp) => (
-  <>
-    <DialogClose position="right" size={size}>
-      <IconButton name="cross" variant="ghost" />
+}: DialogProps & SizeProp & PropsWithChildren) => (
+  <div
+    className={twMerge(
+      "flex items-center justify-between",
+      padding && defaultPaddingClassName,
+      className,
+      size
+    )}
+    {...props}
+  >
+    {children}
+    <DialogClose size={size}>
+      <Icon name="cross" size={size ? iconSize[size] : 20} />
```

Comment:
We could apply this change if we default the `size` prop to `lg`
```suggestion
      <Icon name="cross" size={iconSize[size]} />
```